core/templates/pages/preferences-page/preferences-page.component.html

```diff
@@ -42,7 +42,7 @@
       <div class="form-group row">
         <label for="label-target-user-bio" class="col-lg-4 col-md-4 col-sm-4" translate="I18N_PREFERENCES_BIO"></label>
         <div class="col-lg-8 col-md-8 col-sm-8">
-          <textarea id="label-target-user-bio" class="oppia-bio-border protractor-test-user-bio" ng-model="$ctrl.userBio" ng-blur="$ctrl.saveUserBio($ctrl.userBio)" rows="5" maxlength="2000"></textarea>
+          <textarea id="label-target-user-bio" class="oppia-bio-border protractor-test-user-bio" ng-model="$ctrl.userBio" ng-model-options="{ debounce: 1000 }" ng-change="$ctrl.saveUserBio($ctrl.userBio)" rows="5" maxlength="2000"></textarea>
```

Comment:
Could you explain the added code for a bit, also I think that raising the value to 1500 or 2000 would be better.

Reply:
The following Directives were added:
1. ng-change : event is triggered at every change in the value.
2. debounce : The ng-change will not call the function until 1000 ms after the last key stroke.

I feel 1000ms was long enough. Since I felt the "saved!" alert already took considerable time to appear.

Reply:
Hmm, okay, one more thing, what if the user is on a slow connection, and the new save is triggered even before the last save was completed? How it would be handled? Is there a possibility to add some timeouts between the requests?

Reply:
@vojtechjelinek 
If a user does not type for more than 1 sec then the content would be saved. If a user decides to type after every 1 sec, then the new save is triggered even before the last save was completed. But the saving would occur in a sequential order, i.e., only if the previous request passes or fail will the next one be triggered. 
I also tested with a 1kbs connection (using Chrome dev tools) and it still worked since the edits would be a couple of kb if it's a big change and in Bytes if it is a small one.

Adding timeouts would bring back the original problem. If a user was to reload before the timeout got over the changes would not be saved.

Reply:
Okay, great.